src/OpenCensus.Collector.Dependencies/DependenciesCollectorEventSource.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="DependenciesCollectorEventSource.cs" company="OpenCensus Authors">
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenCensus.Collector.Dependencies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Tracing;
+    using System.Globalization;
+    using System.Text;
+    using System.Threading;
+
+    /// <summary>
+    /// EventSource listing ETW events emitted from the project.
+    /// </summary>
+    [EventSource(Name = "OpenCensus.Collector.Dependencies")]
+    internal class DependenciesCollectorEventSource : EventSource
+    {
+        internal static DependenciesCollectorEventSource Log = new DependenciesCollectorEventSource();
+
+        [NonEvent]
+        public void ExceptionInCustomSampler(Exception ex)
+        {
+            if (Log.IsEnabled(EventLevel.Error, EventKeywords.All))
```

Comment:
this check `IsEnabled` should check for `Warning` level as defined in the actual method. This is one of that hard to catch issues:

```
[Event(2, Message = "Error getting custom sampler, the default sampler will be used. Exception : {0}", Level = EventLevel.Warning)] 
        public void ExceptionInCustomSampler(string ex)
```

Reply:
Done, Thanks ! This is bad copy-paste on my part.